cogs/core.py

```diff
@@ -0,0 +1,386 @@
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <https://www.gnu.org/licenses/>.
+"""
+
+import discord
+
+from datetime import datetime as dt
+from discord.utils import format_dt
+from discord.ext import commands
+from discord import app_commands
+
+from src.errors.checks import DisabledCommand
+from src.ServerUtils import Utils
+
+class Core(commands.Cog):
+    """
+    Main Cog
+
+    Holds all the commands that do not fit a specific catagory
+    """
+
+    def __init__(self, bot):
+        self.bot = bot
+
+    info_group = app_commands.Group(
+        name="info", description="Gives information about server and member."
+    )
+
+    tutorials_group = app_commands.Group(
+        name="tutorials", description="Holds all the tutorials comand"
+    )
+
+    async def cog_load(self):
+        self.bot.lj.log("emperor.cogs.core", "Core cog was loaded")
+
+    async def cog_unload(self):
+        self.bot.lj.warn("emperor.cogs.core", "Core cog was unloaded")
+
+    async def cog_command_error(self, ctx, error):
+        print(error)
+
+    async def cog_app_command_error(self, interaction, error):
+        self.bot.lj.warn(
+            f"emperor.cogs.maincog.{interaction.command.name}",
+            f"<@{interaction.user.id}>, {error}",
+        )
+        
+        if isinstance(error, DisabledCommand):
+            await interaction.response.send_message(
+                f"{error.args}\nPlease ask staff to enable, if need be.",
+                ephemeral=True
+            )
+            return
+
+        
+        await interaction.response.send_message(f"<@{interaction.user.id}>, {error}")
+
+    # Implementing on and off for commands
+    def is_on():
+        def _extras_json() -> dict:
+            return Utils.read_from_json("res/json/extras.json")
+        
+        def predicate(interaction: discord.Interaction, bot: discord.ext.commands.Bot) -> bool:
+            
+            if interaction.command.extras['command_signature'] in _extras_json()['disabled_commands']:
+                raise DisabledCommand("Command has been disabled, therefore cannot be executed.")
+            else:
+                return True
+        
+        async def decorator(interaction: discord.Interaction):
+            
+            bot = interaction.client
+            
+            return predicate(interaction, bot)
+        
+        return app_commands.check(decorator)
+
+
+    async def __count_online_members(self, guild: discord.Guild) -> int:
+        """Counts the number of online members in the guild
+
+        Args:
+            guild (discord.Guild): The guild to get the members from
+
+        Returns:
+            int: The number of online (online & dnd) members
+        """
+
+        # Go through the guild members
+        # Check if their status is dnd/online
+        # increment if so
+        num = 0
+
+        possibleStatus = [
```

Comment:
convert possibleStatus into hashSet so when you do "if member.status in possibleStatus:" it will run in constant time not linear

Reply:
Noted, will be added into to-do list for v1.0.0 merge.